### DISCUSSION TITLE
ASTRIS II (RANDEV)

### DISCUSSION THREAD

**Discussion by DinaGad** [daniestevez/gr-satellites#325] [Satellite teams]
Dear all,
This is Gadisa is a graduate student in the aerospace engineering department. I was recently assigned to work on the communication subsystem of the CubeSat (RANDEV) project, which will be launched in August 2022. I'm presently attempting to understand the working principles of the CubeSat communication subsystem, as well as the gr-satellites file, in order to use it for our ground station.
I need your advice and recommendations to learn gr-satellite files and other related topics as soon as possible so that I can adapt for our ground station because I am short on time and have no background in communication.
@daniestevez, Thank you so much for doing such an excellent job. This satellite project is a continuation of K2SAT, and I learned that you and Pilow, a former KAIST student, worked on GRC and ground station implementation. I'm currently testing our CubeSat's S-band component, but I'm still having issues with GRC and the step of sending and receiving data through the S-band for component functionality testing.
I'd appreciate your assistance and suggestions in testing S-band and implementing GRC so that our ground station can be operational as soon as possible.
Thanks

**Comment by daniestevez** [comment]
Hi Gadisa,

Thanks for reaching out. Can you give us more information about this satellite, specifically, what kind of modulation and protocols it uses? You can link to the relevant project documentation or attach it here.

Is this an amateur satellite? I didn't see it in the [IARU Satellite Coordination page](http://www.amsatuk.me.uk/iaru/index.php).

**Comment by DinaGad** [comment]
Hello Daniel,
Thank you for your inquiry about the satellite. In terms of the satellite's communication component, the S-band employs QPSK, and the specifications and protocols for both the S-band and the TRXVU are attached.
Regarding frequency coordination, we have already submitted the necessary documentation to Korea's National Radio Agency and are awaiting approval.
![S-band_encoding_and_modulation](https://user-images.githubusercontent.com/58734559/150443038-591e75f2-12d0-4c06-b537-14c66fa25fdf.JPG)
![TRXVU_downlink](https://user-images.githubusercontent.com/58734559/150443040-860f4271-d1d5-41ec-a3b2-f4ca737732f7.JPG)
![TRXVU_uplink](https://user-images.githubusercontent.com/58734559/150443042-f3433fb7-3562-4e3e-ac19-6ec0bab439a1.JPG)
[BR-01-00073 Rev C (2).pdf](https://github.com/daniestevez/gr-satellites/files/7909380/BR-01-00073.Rev.C.2.pdf)
Once again, thanks

**Reply by daniestevez** [reply]
Regarding the TRXVU, that's a standard AX.25 packet radio transceiver, and is supported by gr-satellites.

Regarding S-band, that transmitter seems quite similar to the one used in K2SAT. We have blocks for everything you listed, except for OQPSK. If I remember correctly, the framing used by K2SAT S-band was rather odd and inconvenient, and your documentation doesn't mention anything about framing.

**Comment by daniestevez** [comment]
Hi Gadisa,

Will this satellite use the Amateur Satellite service? If so, you should request IARU coordination. Even if not mandatory according to your country's regulations, it is good etiquette with the amateur radio community.

**Comment by DinaGad** [comment]
Hello Daniel,
Regarding frequency coordination, we requested the regional authority Republic of Korea Air Force Academy, which is also assisting in the coordination of the K2SAT, and we verified that they had already sent our file to ITU rather than IARU, and we will take your suggestion into account. 
However, I'm having trouble testing the S-band module using the GRC GNU radio flow-graph. The problem is that I wanted to test the functionality of the S-band module before assembly, but I don't know how to do so with these GRC files. It would be greatly appreciated if you could point me in the right direction. 
Thank you.

**Reply by daniestevez** [reply]
The flowgraph in `examples/satellites/k2sat_image.grc` doesn't have many things to tweak. It expects a 2 Msps IQ recording with the 500 kbaud QPSK signal in the centre. However it will only work if the framing is exactly the same as that used by K2SAT, of which I'm not so certain.
